File: exercises/day13-Keyboard-First-Refactoring/java/src/main/java/sleightrip/trip/SleighTripDAO.java
package sleightrip.trip;

import java.util.List;

import sleightrip.exception.CollaboratorCallException;
import sleightrip.elf.Elf;

/**
 * Legacy static data access object for sleigh trips.
 */
public class SleighTripDAO {

    public static List<SleightTrip> findTripsByUser(Elf user) {
        throw new CollaboratorCallException(
                "SleighTripDAO is a static collaborator and should not be used directly in unit tests"
        );
    }
}

File: exercises/day13-Keyboard-First-Refactoring/kotlin/src/main/kotlin/sleightrip/trip/SleighTripDAO.kt
package sleightrip.trip

import sleightrip.exception.CollaboratorCallException
import sleightrip.elf.Elf

/**
 * Legacy static data access object for sleigh trips.
 */
object SleighTripDAO {
    
    fun findTripsByUser(user: Elf): List<SleightTrip> {
        throw CollaboratorCallException(
            "SleighTripDAO is a static collaborator and should not be used directly in unit tests"
        )
    }
}
